cmd/snap/cmd_info.go

```diff
@@ -179,58 +178,13 @@ func wrapLine(out io.Writer, text []rune, pad string, termWidth int) error {
 		// Rather than let that happen, give up.
 		indent = pad + "  "
 	}
-	return wrapGeneric(out, text, indent, indent, termWidth)
+	return strutil.WordWrap(out, text, indent, indent, termWidth)
 }
 
 // wrapFlow wraps the text using yaml's flow style, allowing indent
 // characters for the first line.
 func wrapFlow(out io.Writer, text []rune, indent string, termWidth int) error {
-	return wrapGeneric(out, text, indent, "  ", termWidth)
-}
-
-// wrapGeneric wraps the given text to the given width, prefixing the
-// first line with indent and the remaining lines with indent2
-func wrapGeneric(out io.Writer, text []rune, indent, indent2 string, termWidth int) error {
-	// Note: this is _wrong_ for much of unicode (because the width of a rune on
-	//       the terminal is anything between 0 and 2, not always 1 as this code
-	//       assumes) but fixing that is Hard. Long story short, you can get close
-	//       using a couple of big unicode tables (which is what wcwidth
-	//       does). Getting it 100% requires a terminfo-alike of unicode behaviour.
-	//       However, before this we'd count bytes instead of runes, so we'd be
-	//       even more broken. Think of it as successive approximations... at least
-	//       with this work we share tabwriter's opinion on the width of things!
-
-	// This (and possibly printDescr below) should move to strutil once
-	// we're happy with it getting wider (heh heh) use.
-
-	indentWidth := utf8.RuneCountInString(indent)
-	delta := indentWidth - utf8.RuneCountInString(indent2)
-	width := termWidth - indentWidth
-
-	// establish the indent of the whole block
-	var err error
-	for len(text) > width && err == nil {
-		// find a good place to chop the text
-		idx := runesLastIndexSpace(text[:width+1])
```

Comment:
This can now go in the `cmd/snap` package too, yes? Right now it seems we would end up with two versions of `runesLastIndexSpace()` of which one is unused(?)

Reply:
Yes, true, we end up with two of them. But both are used, cmd_info has a more specific-purpose `wrapLine` for yaml function as well which also calls `runesLastIndexSpace` that now uses `strutil.WordWrap` (underlying) as well. I did not want to move `wrapLine` into strutil, and the `runesLastIndexSpace` seems to simple to expose (maybe that is just me?) in this case, or atleast I felt it a bit overkill, but I'll happily change it

Reply:
Sorry, reading my comment it was vague and unclear. To be precise, doing this:
```go
diff --git a/cmd/snap/cmd_info.go b/cmd/snap/cmd_info.go
index db3eafd3df..32b8643eb7 100644
--- a/cmd/snap/cmd_info.go
+++ b/cmd/snap/cmd_info.go
@@ -147,17 +147,6 @@ func runesTrimRightSpace(text []rune) []rune {
        return text[:j]
 }
 
-// runesLastIndexSpace returns the index of the last whitespace rune
-// in the text. If the text has no whitespace, returns -1.
-func runesLastIndexSpace(text []rune) int {
-       for i := len(text) - 1; i >= 0; i-- {
-               if unicode.IsSpace(text[i]) {
-                       return i
-               }
-       }
-       return -1
-}
-
 // wrapLine wraps a line, assumed to be part of a block-style yaml
 // string, to fit into termWidth, preserving the line's indent, and
 // writes it out prepending padding to each line.
```
does not cause any test to break in your branch so I think we can remove the version of runesLastIndexSpace in the `snap` package.

Reply:
Nope I made a mistake, you are completely correct, I relied on go to inform me that I have an unused function which I can see it does in fact not in this case, so I automatically assumed it was used in lineWrap! I'm sorry for the miscommunication